### PR TITLE
Fix mac popup special keys

### PR DESCRIPTION
**Pull request by ltetak**
## What does the pull request do?
Special keys don't work in Popups on macOS. (Special key is anything written with modifier - even shift+1 for bang sign)

## What is the current behavior?
InputManager is not initialized. But that's correct because the Window is still the key window and it receives the hotkeys.

## What is the updated/expected behavior with this PR?
Keep the InputManager initialized to the parent window.

## How was the solution implemented (if it's not obvious)?
I don't know if this is the correct solution in the scope of all platforms.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Breaking changes
test on other platforms

## Fixed issues
https://github.com/AvaloniaUI/Avalonia/issues/14408

### CODE REVIEW

**Comment by Gillibald** [comment]
Popups under macOS currently don't implement any input method

**Comment by ltetak** [comment]
@Gillibald but they can use the parent window one, can't they?

**Comment by Gillibald** [comment]
> @Gillibald but they can use the parent window one, can't they?

No Popup itself needs to process the input context messages

**Comment by ltetak** [comment]
@Gillibald I am not familiar that much with native mac code. Can you help me what I am missing here? (last commit here)
I believe the popup must be key window, right? And it must become firstResponder upon show too.
The problem is that when I click on the popup in this case it closes immediately. I can't figure out what is the problem.
This is a big problem for us because textboxes in popups do not work.

File: src/Avalonia.Base/Threading/Dispatcher.Queue.cs [hunk 1]
```diff
@@ -194,7 +194,6 @@ void ExecuteJobsCore(bool fromExplicitBackgroundProcessingCallback)
                 
                 if (Now - backgroundJobExecutionStartedAt.Value > _maximumInputStarvationTime)
                 {
-                    _signaled = true;
```
Comment (Gillibald) [inline src/Avalonia.Base/Threading/Dispatcher.Queue.cs:197]: Unrelated change?

Comment (ltetak) [reply src/Avalonia.Base/Threading/Dispatcher.Queue.cs:197]: Yes. I need to rebase but for some reasons this must wait :) ignore please for now

**Comment by ltetak** [comment]
@Gillibald I am not sure how to do it properly though. I can override `ShouldTakeFocusOnShow` or I can make it first responder manually. In either case, the main window loses the focus and the flyout is hidden. This is the problem even now (if you click the popup it closes immediately). I couldn't find the connection between being a key window and closing on getting focus. Can you point me to the right direction here? Thanks!
